cmd/create/machinepool/cmd.go

```diff
@@ -17,255 +17,100 @@ limitations under the License.
-		"max-replicas",
-		0,
-		"Maximum number of machines for the machine pool.",
-	)
-
-	flags.StringVar(
-		&args.InstanceType,
-		"instance-type",
-		"m5.xlarge",
-		"Instance type that should be used.",
-	)
-
-	flags.StringVar(
-		&args.Labels,
-		"labels",
-		"",
-		"Labels for machine pool. Format should be a comma-separated list of 'key=value'. "+
-			"This list will overwrite any modifications made to Node labels on an ongoing basis.",
-	)
-
-	flags.StringVar(
-		&args.Taints,
-		"taints",
-		"",
-		"Taints for machine pool. Format should be a comma-separated list of 'key=value:ScheduleType'. "+
-			"This list will overwrite any modifications made to Node taints on an ongoing basis.",
-	)
-
-	flags.BoolVar(
-		&args.UseSpotInstances,
-		"use-spot-instances",
-		false,
-		"Use spot instances for the machine pool.",
-	)
-
-	flags.StringVar(
-		&args.SpotMaxPrice,
-		"spot-max-price",
-		"on-demand",
-		"Max price for spot instance. If empty use the on-demand price.",
-	)
-
-	flags.BoolVar(
-		&args.MultiAvailabilityZone,
-		"multi-availability-zone",
-		true,
-		"Create a multi-AZ machine pool for a multi-AZ cluster")
-
-	flags.StringVar(
-		&args.AvailabilityZone,
-		"availability-zone",
-		"",
-		"Select availability zone to create a single AZ machine pool for a multi-AZ cluster")
-
-	flags.StringVar(
-		&args.Subnet,
-		"subnet",
-		"",
-		"Select subnet to create a single AZ machine pool for BYOVPC cluster")
-
-	flags.StringVar(
-		&args.Version,
-		"version",
-		"",
-		"Version of OpenShift that will be used to install a machine pool for a hosted cluster,"+
-			" for example \"4.12.4\"",
-	)
+type CreateMachinePool struct {
+	service machinepool.MachinePoolService
+}
 
-	flags.BoolVar(
-		&args.Autorepair,
-		"autorepair",
-		true,
-		"Select auto-repair behaviour for a machinepool in a hosted cluster.",
-	)
+func NewCreateMachinePool(spec CreateMachinePoolSpec) CreateMachinePool {
+	return CreateMachinePool{
+		service: spec.Service,
+	}
+}
 
-	flags.StringVar(
-		&args.TuningConfigs,
-		"tuning-configs",
-		"",
-		"Name of the tuning configs to be applied to the machine pool. Format should be a comma-separated list. "+
-			"Tuning config must already exist. "+
-			"This list will overwrite any modifications made to node tuning configs on an ongoing basis.",
-	)
+func NewCreateMachinePoolCommand() *cobra.Command {
+	cmd, options := mpOpts.BuildMachinePoolCreateCommandWithOptions()
+	cmd.Run = rosa.DefaultRunner(rosa.RuntimeWithOCM(), CreateMachinepoolRunner(options))
+	return cmd
+}
 
-	flags.StringVar(
-		&args.KubeletConfigs,
-		"kubelet-configs",
-		"",
-		"Name of the kubelet config to be applied to the machine pool. A single kubelet config is allowed. "+
-			"Kubelet config must already exist. "+
-			"This will overwrite any modifications made to node kubelet configs on an ongoing basis.",
-	)
+// Create machine pool based on cluster type
+func (m CreateMachinePool) createMachinePoolBasedOnClusterType(r *rosa.Runtime,
```

Comment:
@den-rgb I kinda feel like this should be in the `service`. So we have a single method on the `service` which is `createMachinePool` (Machine Pool is the consistent terminology) and internally, the service works out whether it needs to create a Classic MachinePool or a HCP Machine pool.

WDYT?

Reply:
Yeah good idea I'll move it over